src/IO/GEHDF5Wrapper.cxx

```diff
@@ -383,31 +401,40 @@ Succeeded GEHDF5Wrapper::initialise_geo_factors_data(const std::string& path,
 
     return Succeeded::yes;
 }
-
+// This info is in norm3d file
 Succeeded GEHDF5Wrapper::initialise_efficiency_factors(const std::string& path)
 {
     if(path.size() == 0)
     {
-        if(is_signa)
+        if(is_signa)  // AB todo "is_RDF9"
         {
+            
             m_address = "3DCrystalEfficiency/crystalEfficiency";
+            // Get the DataSpace (metadata) corresponding to the DataSet that we want to read
             m_dataset_sptr.reset(new H5::DataSet(file.openDataSet(m_address)));
             m_dataspace = m_dataset_sptr->getSpace();
-            int rank = m_dataspace.getSimpleExtentNdims();
-            hsize_t dims_out[2];
-           // int ndims = m_dataspace.getSimpleExtentDims( dims_out, NULL);
+            // Create an array to host the size of the dimensions
+            const int rank = m_dataspace.getSimpleExtentNdims();
+            hsize_t dims[rank];
+            // hsize_t max_dims[dataspace_Ndims]; // AB: Do we want the max_dims?
+            // Read size of dimensions
+            m_dataspace.getSimpleExtentDims( dims, NULL); // AB: Do we want the max_dims?
+
+            // AB: todo, is this debugging residue? should we just remove?
+            // int ndims = m_dataspace.getSimpleExtentDims( dims_out, NULL);
                  std::cout << "rank " << rank << ", dimensions " <<
-                     (unsigned long)(dims_out[0]) << " x " <<
-                     (unsigned long)(dims_out[1]) << std::endl;
-            //! \todo Get these numbers from the HDF5 file
-            {
-                m_NX_SUB = 45;    // hyperslab dimensions
-                m_NY_SUB = 448;
-                m_NZ_SUB = 1;
-                m_NX = 45;        // output buffer dimensions
-                m_NY = 448;
-                m_NZ = 1;
-            }
+                     (unsigned long)(dims[0]) << " x " <<
+                     (unsigned long)(dims[1]) << std::endl;
+
```

Comment:
The dimensions of this crystal efficiency in the actual norm3d file is [45 896]
GE repeats the values twice. Whereas, we need to read part of that array :[45 448]
We need to extract the crystal efficiency factors for ring and detector numbers. We then use the same array to calculate the normalisation factors.

Reply:
@pw351 Do you have any idea if this is the standard for RDF9 or just Signa? say, should I just grab the dims and half its value? because we only want to read half of it?

Reply:
let's read half. if scanner is already initialised (should be I guess), add a check that you then get `scanner_sptr->get_num_detectors_per_ring()`